docs/pages/enroll-resources/workload-identity/aws-oidc-federation.mdx

```diff
@@ -0,0 +1,431 @@
+- OIDC Federation requires that your Teleport Proxy Service is reachable by
+  AWS, whereas Roles Anywhere does not.
+
+This guide covers configuring OIDC federation, for Roles Anywhere, see
+[Configuring Workload Identity and AWS Roles Anywhere](./aws-roles-anywhere.mdx).
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- `tbot` must already be installed and configured on the host where the
+workloads which need to access Teleport Workload Identity will run. For more
+information, see the [deployment guides](../machine-id/deployment.mdx).
+
+<Notice type="warning">
+Issuing JWT SVIDs with Teleport Workload Identity requires at minimum version
+16.4.3.
+</Notice>
+
+### Deciding on a SPIFFE ID structure
+
+Within Teleport Workload Identity, all identities are represented using a
+SPIFFE ID. This is a URI that uniquely identifies the entity that the identity
+represents. The scheme is always `spiffe://`, and the host will be the name of
+your Teleport cluster. The structure of the path of this URI is up to you.
+
+For the purposes of this guide, we will be granting access to AWS to the
+`spiffe://example.teleport.sh/svc/example-service` SPIFFE ID.
+
+If you have already deployed Teleport Workload Identity, then you will already
+have a SPIFFE ID structure in place. If you have not, then you will need to
+decide on a structure for your SPIFFE IDs.
+
+If you are only using Teleport Workload Identity with AWS OIDC Federation, you
+may structure your SPIFFE IDs so that they explicitly specify the AWS role they
+are allowed to assume. However, it often makes more sense to name the workload
+or person that will use the SPIFFE ID. See the
+[best practices guide](./best-practices.mdx) for further advice.
+
+## Step 1/4. Configure AWS
+
+Configuring AWS OIDC Federation for the first time involves a few steps. Some of
+these may not be necessary if you have previously configured AWS OIDC Federation
+for your Teleport cluster.
+
+### Create an OpenID Connect Identity Provider
+
+First, you'll need to create an OIDC identity provider in AWS. This will define
+a trust relationship between AWS and your Teleport cluster's Workload Identity
+issuer. You can reuse this OIDC identity provider to grant different workloads
+using Teleport Workload Identity access to AWS services.
+
+When configuring the provider, you need to specify the issuer URI. This will be
+the public address of your Teleport Proxy Service with the path
+`/workload-identity` appended. Your Teleport Proxy Service must be accessible
+by AWS in order for OIDC federation to work.
+
+<Tabs>
+<TabItem label="Terraform">
+
+Before you can configure the OIDC identity provider, you need to determine the
+thumbprint of the TLS certificate used by your Teleport Proxy Service. You can
+do this using `curl`:
+
+```code
+$ curl https://example.teleport.sh/webapi/thumbprint
+"example589ee4bf31a11b78c72b8d13f0example"%
+```
+
```

Comment:
Let's add a transition between the command and the HCL snippet so it's more explicit what a reader should do.

Reply:
a29d4c92ad